resources/crds/benchmark_v1alpha1_smallfile.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: benchmark.example.com/v1alpha1
+kind: Benchmark
+metadata:
+  name: example-benchmark
+  namespace: ripsaw
+spec:
+  workload:
+    name: smallfile
+    args:
+      clients: 1
+      operation: create
+      threads: 1
+      file-size: 4
+      files: 100
+      top: /tmp/smallfile/
+      fsync: y
```

Comment:
I don't think you want fsync: y.    This will cause an fsync after every file create, which is very expensive and not the typical use case.    Also, you want to add these parameters to improve your data collection:

```
args:
...
   threads: 1
   response-times: Y
   output-json:  <wherever you want to put it>
```

output-json means that you don't have to screen-scrape to get results from smallfile.    

response-times: Y will cause each smallfile thread (in each pod) to save response times for all of its file operations in $top/network_shared/rsptimes_<thread_id>_<host>_<operation>_<time>.csv .   The smallfile_rsptimes_stats.py program postprocesses those response times into response time percentiles, including percentiles as a function of time.

threads: 1 means that only 1 thread per pod will be run.   Perhaps we want this to be user-settable but typically you can get more concurrency by just adding more pods, and this has the advantage of letting Kubernetes spread your threads across the available physical hosts more evenly.

Reply:
I agree to the idea of 1 thread per pod, which we followed for fio as well. I will omit this and add another patch. 

For  gluster use case, we generally use to do a fsync after every run to clear the buffers to create more stable run ( when using fio). Hence, in smallfile , i thought it would be good to add fsync after the run. If you still think , fsync is not need, then i can clear it out. 

output-json, I am already using this option in the cli command, and it is capturing the results at the desired location.

Reply:
In smallfile, "--fsync Y" means that it issues an [fsync call before each file is closed](https://github.com/distributed-system-analysis/smallfile/blob/master/smallfile.py#L1089).   While this can sometimes be a useful test for debugging certain performance problems, it is not what you want to do by default.    I assume that smallfile parameters can be added to the CR yaml if needed.